artichoke-backend/src/extn/core/array/array.rb

```diff
@@ -1105,12 +1105,20 @@ def select!(&block)
     replace(result)
   end
 
-  def shuffle(_rng = (not_set = true)) # rubocop:disable Lint/UselessAssignment
-    raise NotImplementedError, 'TODO implement in Rust'
+  def shuffle(rng = nil)
+    rng_instance = rng.nil? ? Random.new : rng
+    shuffled_orders = (0..size - 1).map.with_index { |_n, i| [rng_instance.rand, i] }.sort { |a, b| a.first <=> b.first }
```

Comment:
This suggestion adds support for `[].shuffle` using the default `Random` instance and makes a few performance optimizations. Let me know what you think @KBuild.

```suggestion
  def shuffle(rng = (not_set = true))
    rng = Random::DEFAULT if not_set
    shuffled_orders = (0...size).map { |idx| [rng.rand, idx] }.sort { |a, b| a[0] <=> b[0] }
```

Reply:
Thank you for informing suggestion!
I prefer the code keeping consistency rules
I will fix that

Reply:
Oh, I check some document and find `Array#shuffle` in this link
https://docs.ruby-lang.org/en/2.6.0/Array.html#method-i-shuffle

It seems like to using `random` keyword arguments
So I think we should use

```rb
def shuffle(random: (not_set = true))
...
end
def shuffle(random: (not_set = true))
...
end
```

How about this?

I'm going to commit this idea, and please check code 🙏

Reply:
Looks good. Thanks for checking the docs.